nixos/modules/services/networking/firezone/gui-client.nix

```diff
@@ -0,0 +1,141 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    boolToString
+    getExe'
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+
+  cfg = config.services.firezone.gui-client;
+in
+{
+  options = {
+    services.firezone.gui-client = {
+      enable = mkEnableOption "the firezone gui client";
+      package = mkPackageOption pkgs "firezone-gui-client" { };
+
+      allowedUsers = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = ''
+          All listed users will become part of the `firezone-client` group so
+          they can control the IPC service. This is a convenience option.
+        '';
+      };
+
+      name = mkOption {
+        type = types.str;
+        description = "The name of this client as shown in firezone";
+      };
+
+      logLevel = mkOption {
+        type = types.str;
+        default = "info";
+        description = ''
+          The log level for the firezone application. See
+          [RUST_LOG](https://docs.rs/env_logger/latest/env_logger/#enabling-logging)
+          for the format.
+        '';
+      };
+
+      enableTelemetry = mkEnableOption "telemetry";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups.firezone-client.members = cfg.allowedUsers;
+
+    # Required for deep-link mimetype registration
+    environment.systemPackages = [ cfg.package ];
+
+    # Required for the token store in the gui application
+    services.gnome.gnome-keyring.enable = true;
+
+    systemd.services.firezone-ipc-service = {
+      description = "GUI IPC service for the Firezone zero-trust access platform";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      path = [ pkgs.util-linux ];
+      script = ''
+        # If FIREZONE_ID is not given by the user, use a persisted (or newly generated) uuid.
+        if [[ -z "''${FIREZONE_ID:-}" ]]; then
+          if [[ ! -e client_id ]]; then
+            uuidgen -r > client_id
+          fi
+          export FIREZONE_ID=$(< client_id)
+        fi
+
+        exec ${getExe' cfg.package "firezone-client-ipc"} run
+      '';
+
+      environment = {
+        FIREZONE_NAME = cfg.name;
+        FIREZONE_NO_TELEMETRY = boolToString (!cfg.enableTelemetry);
```

Comment:
This doesn't do anything, the GUI client doesn't have a switch to turn off telemetry. (Telemetry is always off for self-hosted instances)

Reply:
Oh good point, I probably copy pasted this from the headless client